intents/intent.ridl

```diff
@@ -0,0 +1,183 @@
+webrpc = v1
+
+name = sequence-waas-intents
+version = v0.1.0
+
+struct Intent
+  - version: string
+  - name: string
+  - expiresAt: uint64
+  - issuedAt: uint64
+  - data: any
+  - signatures: []Signature
+    + go.tag.json = ,omitempty
+
+struct Signature
+  - sessionId: string
+     + go.field.name = SessionID
+  - signature: string
+
+# no way to generate string enums
+# enum IntentName: string
+#  - openSession
+#  - closeSession
+#  - validateSession
+#  - finishValidateSession
+#  - listSessions
+#  - getSession
+#  - signMessage
+#  - sendTransaction
+
+struct IntentDataOpenSession
+  - sessionId: string
+    + go.field.name = SessionID
+  - email?: string
+    + go.tag.json = ,omitempty
+  - idToken?: string
+    + go.tag.json = ,omitempty
+
+struct IntentDataCloseSession
+  - sessionId: string
+    + go.field.name = SessionID
+
+struct IntentDataValidateSession
+  - sessionId: string
+    + go.field.name = SessionID
+  - wallet: string
+  - deviceMetadata: string
+
+struct IntentDataFinishValidateSession
+  - sessionId: string
+    + go.field.name = SessionID
+  - wallet:  string
+  - salt: string
+  - challenge: string
+
+struct IntentDataListSessions
+  - wallet: string
+
+
+struct IntentDataGetSession
+  - sessionId: string
+    + go.field.name = SessionID
+  - wallet: string
+
+struct IntentDataSignMessage
+  - network: string
+  - wallet:  string
+  - message: string
+
+struct IntentDataSendTransaction
+  - network: string
+  - wallet:  string
+  - identifier: string
```

Comment:
what is identifier btw..? maybe you can just add `# identifier is.............` comment above it..? I forget again

Reply:
It's used to generate nonce space.